### PR TITLE
[User Model] [Fix] Erroring on User property updates before push subscription create

### PR DESCRIPTION
**Pull request by jkasten2**
# Description
## 1 Line Summary
Fixes an error where any User property update before a push subscribed is crated causes the SDK to throw.

## Details

# Validation
## Tests
* Added new automated test
* Manually tested by calling `OneSignal.User.addTag("key", "value");` then accepting notifications, ensuring the User create call included the tags.
### Checklist
   - [X] All the automated tests pass or I explained why that is not possible
   - [X] I have personally tested this on my machine or explained why that is not possible
   - [X] I have included test coverage for these changes or explained why they are not needed

**Programming Checklist**
Interfaces:
   - [X] Don't use default export
   - [X] New interfaces are in model files

Functions:
   - [X] Don't use default export
   - [X] All function signatures have return types
   - [X] Helpers should not access any data but rather be given the data to operate on.

Typescript:
   - [X] No Typescript warnings
   - [X] Avoid silencing null/undefined warnings with the exclamation point

Other:
   - [X] Iteration: refrain from using `elem of array` syntax. Prefer `forEach` or use `map`
   - [X] Avoid using global OneSignal accessor for `context` if possible. Instead, we can pass it to function/constructor so that we don't call `OneSignal.context`

### Checklist
   - [X] I have included screenshots/recordings of the intended results or explained why they are not needed

---

## Related Tickets

---

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Website-SDK/1061)

### CODE REVIEW

File: src/core/executors/ExecutorResult.ts [hunk 1]
```diff
@@ -1,7 +1,29 @@
-export default class ExecutorResult<Model> {
-  constructor(
-    readonly success: boolean,
-    readonly retriable: boolean,
-    readonly result?: Model
-    ){}
+export default interface ExecutorResult<Model> {
```
Comment (rgomezp) [inline src/core/executors/ExecutorResult.ts:1]: no longer need to export

File: src/core/requestService/IdentityRequests.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import OneSignalApiBaseResponse from "../../shared/api/OneSignalApiBaseResponse";
 import { logMethodCall } from "../../shared/utils/utils";
 import OneSignalError from "../../shared/errors/OneSignalError";
-import ExecutorResult from "../executors/ExecutorResult";
+import ExecutorResult, {
```
Comment (rgomezp) [inline src/core/requestService/IdentityRequests.ts:4]: unneeded import

File: src/core/requestService/SubscriptionRequests.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
 import MainHelper from "../../shared/helpers/MainHelper";
 import OneSignalError from "../../shared/errors/OneSignalError";
 import { logMethodCall } from "../../shared/utils/utils";
-import ExecutorResult from "../executors/ExecutorResult";
+import ExecutorResult, {
```
Comment (rgomezp) [inline src/core/requestService/SubscriptionRequests.ts:4]: unneeded import

File: src/core/requestService/UserPropertyRequests.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
 import { logMethodCall } from "../../shared/utils/utils";
 import OneSignalError from "../../shared/errors/OneSignalError";
-import ExecutorResult from "../executors/ExecutorResult";
+import ExecutorResult, {
```
Comment (rgomezp) [inline src/core/requestService/UserPropertyRequests.ts:3]: unneeded import

**Review by rgomezp — CHANGES_REQUESTED** [review]

File: src/core/requestService/IdentityRequests.ts (hunk 2 above)
Comment (jkasten2) [reply src/core/requestService/IdentityRequests.ts:4]: It's still being used as below as as type see this line:
```typescript
static async addIdentity<Model>(operation: Operation<Model>): Promise<ExecutorResult<IdentityModel>> {
```

File: src/core/executors/ExecutorResult.ts (hunk 1 above)
Comment (jkasten2) [reply src/core/executors/ExecutorResult.ts:1]: This still needs to be exported as it is still a result type used outside of this file.
But it should probably don't be `default`, so I can fix that.

Comment (jkasten2) [reply src/core/executors/ExecutorResult.ts]: Removed `default` here

File: src/core/requestService/SubscriptionRequests.ts (hunk 3 above)
Comment (jkasten2) [reply src/core/requestService/SubscriptionRequests.ts]: Import is need in this file, but just removed the 'default' import here.

File: src/core/requestService/UserPropertyRequests.ts (hunk 4 above)
Comment (jkasten2) [reply src/core/requestService/UserPropertyRequests.ts]: Import is need in this file, but just removed the 'default' import here.

**Review by rgomezp — APPROVED** [review]

**Pull request merged**